Mathlib/NumberTheory/NumberField/ProductFormula.lean

```diff
@@ -0,0 +1,110 @@
+/-
+Copyright (c) 2024 Fabrizio Barroero. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Fabrizio Barroero
+-/
+import Mathlib.NumberTheory.NumberField.FinitePlaces
+
+/-!
+# The Product Formula for number fields
+In this file we prove the Product Formula for number fields: for any non-zero element `x` of a
+number field `K`, we have `∏ |x|ᵥ=1` where the product runs over the equivalence classes of absolute
+values of `K` and the `|⬝|ᵥ` are suitably normalized.
+
+## Main Results
+* `NumberField.FinitePlace.prod_eq_inv_abs_norm`: for any non-zero element `x` of a number field
+`K`, the product `∏ |x|ᵥ` of the absolute values of `x` associated to the finite places of `K` is
+equal to the inverse of the norm of `x`.
+* `NumberField.prod_abs_eq_one`: for any non-zero element `x` of a number field `K`, we have
+`∏ |x|ᵥ=1`, where the product runs over the equivalence classes of absolute values of `K`.
+
+## Tags
+number field, embeddings, places, infinite places, finite places, product formula
+-/
+
+namespace NumberField
+
+variable {K : Type*} [Field K] [NumberField K]
+
+open Algebra Classical FinitePlace Function Ideal IsDedekindDomain
+  IsDedekindDomain.HeightOneSpectrum
```

Comment:
```suggestion
open Algebra
```
This suggestion seems to have disappeared. (The idea is to llocalize the `open` statements to where they are actually needed.)

Reply:
I was actually wondering about that. I am making the changes now...
Thanks for the review! I am learning a lot!